### ISSUE TITLE
Why is Roaring64Map::write() not equal to Roaring64Map.getSizeInBytes()?

### ISSUE THREAD

**Issue by WeeJang** [RoaringBitmap/CRoaring#83]
Hi, I meet a question, do I use API incorrectly ? 
I want to dump the Roaring64Map instance to disk, so my code is ：
``` 
 void dump_to_file(Roaring64Map& bit_map,std::string dump_file_path){
                auto expected_size = bit_map.getSizeInBytes() ;
                char* serialized_bytes = new char[expected_size];
                auto write_ret = bit_map.write(serialized_bytes);
                LOG("expected_size : %llu, write_ret : %llu",expected_size,write_ret);// **output :[expected_size : 12 , write_ret:16 ]**
               .....
               ::write(fd,serialized_bytes,expected_size) ; 
               ......
                delete[] serialized_bytes;
        }
``` 
but it did cored-dump(because **heap-buffer-overflow**).I tested many cases, and found : 
 1) write_ret is always more 4-bytes in length than expected_size;
2) When I replace Roaring , they will be same in length and program run correctly.
Thanks!

**Comment by lemire** [comment]
@ksajme Want to have a look?

**Comment by ksajme** [comment]
Right,

Line 504 of Roaring64Map was changed from the erroneous type of uint32_t to uint64_t during this commit: https://github.com/RoaringBitmap/CRoaring/pull/80

But getSizeInBytes() was not updated with the four byte difference.

Thanks for the bug report!

**Comment by WeeJang** [comment]
👍   @lemire  @ksajme . Thanks !

**Comment by lemire** [comment]
Hopefully addressed by https://github.com/RoaringBitmap/CRoaring/pull/85

**Comment by lemire** [comment]
I am assuming it is now fixed in release 0.2.22. Reopen if you can't verify.
